Python/ceval.c

```diff
@@ -5705,6 +5705,8 @@ _PyEval_EvalFrameDefault(PyThreadState *tstate, _PyInterpreterFrame *frame, int
         _unknown_opcode:
 #else
         EXTRA_CASES  // From opcode.h, a 'case' for each unused opcode
+            /* Load opcode for MSVC to optimize switch(opcode) separately */
+            opcode = _Py_OPCODE(*next_instr);
```

Comment:
IIUC, `TARGET()` macro is not used in `EXTRA_CASES` (and `DO_TRACING`) now. More comments are needed here in case that `EXTRA_CASES` are wrapped with `TARGET()`, or `frame->prev_instr` is needed?

Reply:
Oh you're right. This code is never straightforward... :-( If @markshannon agrees with your analysis let's merge it. Do you have a preferred quick micro-benchmark to show it has any effect at all? Or do you just look at the generated code?

Reply:
Here is a PGO result:
```
Slower (8):
- dulwich_log: 188 ms +- 2 ms -> 208 ms +- 34 ms: 1.11x slower
- nbody: 153 ms +- 2 ms -> 162 ms +- 4 ms: 1.06x slower
- crypto_pyaes: 119 ms +- 2 ms -> 124 ms +- 4 ms: 1.04x slower
- chaos: 131 ms +- 2 ms -> 136 ms +- 2 ms: 1.03x slower
- deltablue: 6.22 ms +- 0.13 ms -> 6.42 ms +- 0.27 ms: 1.03x slower
- xml_etree_parse: 240 ms +- 3 ms -> 247 ms +- 3 ms: 1.03x slower
- sqlalchemy_imperative: 45.3 ms +- 0.6 ms -> 46.6 ms +- 0.8 ms: 1.03x slower
- scimark_sor: 186 ms +- 4 ms -> 190 ms +- 3 ms: 1.02x slower

Faster (17):
- unpack_sequence: 78.6 ns +- 0.3 ns -> 57.5 ns +- 0.5 ns: 1.37x faster
- scimark_lu: 166 ms +- 2 ms -> 147 ms +- 6 ms: 1.13x faster
- logging_silent: 183 ns +- 6 ns -> 164 ns +- 7 ns: 1.12x faster
- tornado_http: 414 ms +- 45 ms -> 372 ms +- 5 ms: 1.11x faster
- xml_etree_iterparse: 203 ms +- 20 ms -> 183 ms +- 11 ms: 1.11x faster
- mako: 19.3 ms +- 0.2 ms -> 17.9 ms +- 0.2 ms: 1.08x faster
- pathlib: 184 ms +- 4 ms -> 174 ms +- 12 ms: 1.06x faster
- regex_compile: 232 ms +- 14 ms -> 220 ms +- 3 ms: 1.05x faster
- fannkuch: 629 ms +- 20 ms -> 602 ms +- 19 ms: 1.04x faster
- xml_etree_process: 107 ms +- 4 ms -> 103 ms +- 1 ms: 1.04x faster
- scimark_sparse_mat_mult: 5.95 ms +- 0.14 ms -> 5.73 ms +- 0.13 ms: 1.04x faster
- sympy_sum: 317 ms +- 22 ms -> 306 ms +- 4 ms: 1.03x faster
- go: 238 ms +- 6 ms -> 231 ms +- 2 ms: 1.03x faster
- spectral_norm: 167 ms +- 3 ms -> 162 ms +- 4 ms: 1.03x faster
- float: 141 ms +- 7 ms -> 137 ms +- 1 ms: 1.03x faster
- telco: 11.3 ms +- 0.1 ms -> 11.0 ms +- 0.2 ms: 1.03x faster
- unpickle: 22.6 us +- 0.3 us -> 22.0 us +- 0.3 us: 1.03x faster

Benchmark hidden because not significant (33): 2to3, chameleon, django_template, hexiom, json_dumps,
 json_loads, logging_format, logging_simple, meteor_contest, nqueens, pickle, pickle_dict, pickle_li
st, pickle_pure_python, pidigits, pyflate, python_startup, python_startup_no_site, raytrace, regex_d
na, regex_effbot, regex_v8, richards, scimark_fft, scimark_monte_carlo, sqlalchemy_declarative, sqli
te_synth, sympy_expand, sympy_integrate, sympy_str, unpickle_list, unpickle_pure_python, xml_etree_g
enerate

Geometric mean: 1.02x faster
```
 IIRC 1-3% faster on Release(Ob2,Ob3) and PGO builds. `unpack_sequence` gets 30-50% faster only when the sequence contains 10 objects or so.